### PR TITLE
Implemented UXTAB16

### PR DESCRIPTION
**Pull request by FernandoS27**

### CODE REVIEW

**Comment by MerryMage** [comment]
Implementation LGTM, but please provide a test for the instruction.

File: src/frontend/translate/translate_arm/extension.cpp [hunk 1]
```diff
@@ -96,8 +96,18 @@ bool ArmTranslatorVisitor::arm_UXTAB(Cond cond, Reg n, Reg d, SignExtendRotation
 }
 
 bool ArmTranslatorVisitor::arm_UXTAB16(Cond cond, Reg n, Reg d, SignExtendRotation rotate, Reg m) {
-    UNUSED(cond, n, d, rotate, m);
-    return InterpretThisInstruction();
+    if (d == Reg::PC || m == Reg::PC || n == Reg::PC)
+        return UnpredictableInstruction();
+    if (ConditionPassed(cond)) {
+        auto rotated = SignZeroExtendRor(m, rotate);
+        auto lower_half = ir.ZeroExtendByteToWord(ir.LeastSignificantByte(rotated));
```
Comment (MerryMage) [inline src/frontend/translate/translate_arm/extension.cpp]: 1. This fails IR type verification.

2. Better:

Replace 

    auto lower_half = ir.ZeroExtendByteToWord(ir.LeastSignificantByte(rotated));
    auto upper_half = ir.And(rotated, ir.Imm32(0x00FF0000));
    auto result = ir.Or(lower_half, upper_half);

with 

    auto result = ir.And(rotated, ir.Imm32(0x00FF00FF));

**Comment by MerryMage** [comment]
Please also provide a test to this function.

**Comment by FernandoS27** [comment]
the test was there, it just wasn't annotated for some reason. Ayways, changes made and tested.

**Pull request merged**